pkg/controller/common/backoff.go

```diff
@@ -0,0 +1,43 @@
+package common
```

Comment:
not 100% sure about this file being needed. If we want to keep it, I think it should be used everywhere we use `wait.Backoff` (I found two other places while quickly grepping through the code), and then it should probably use the same values everywhere, except where there are good reasons to deviate from those defaults. if it's only used in this PR, let's throw it out because it's really just a very thin wrapper around `wait.Backoff`

Reply:
You're right that we use `wait.Backoff` in the code base, but in a little bit different context.

The backoff use cases you are talking about are used just as arguments for function `wait.ExponentialBackoff` which uses this object under the hood and manage its state. On the other hand, the purpose of this wrapper is to calculate and keep backoff value that is not used in the scope of a single function call. I intentionally moved it out from `controlplane` package to hide its implementation details and exported only its essential functions (I wanted to hide locking). Additionally `common.Backoff` is concurrent-safe and might be used by multiple goroutines, and it's not relevant for the rest of uses cases, because those are used in the scope of `wait.ExponentialBackoff`.